### ISSUE TITLE
fuzzy_matching was not present

### ISSUE THREAD

**Issue by srobertson** [mpickering/slack-api#45]
slack-api was working great a week ago now I'm receiving this

```
*** Exception: user error (When parsing the record Preferences of type 
Web.Slack.Types.Preferences.Preferences the key fuzzy_matching was not present.)
```

I tried making _prefFuzzyMatching a Maybe

Now I receive

```
*** Exception: user error (when expecting a Int, encountered Boolean instead)
```

Have no idea which field is giving this.

Any thoughts on making the lib more resilient to upstream changes from slack?

**Comment by traytonwhite** [comment]
For this particular issue: looking at the raw JSON response, it's _teamRequireAtForMention in TeamPreferences (require_at_for_mention in the team.prefs object of the JSON) that has changed to a Boolean.

Agreed the lack of documentation and changing API is frustrating.

**Comment by srobertson** [comment]
that did it? How does one go about debugging that?

**Comment by traytonwhite** [comment]
The error message I was getting had a bit more info:
`user error (Error in $.team.prefs: failed to parse field team: failed to parse field prefs: expected Int, encountered Boolean)`

Then I just manually looked through the JSON response and checked which fields were actually Bool but had Int listed in TeamPreferences.hs.

**Comment by traytonwhite** [comment]
This should be fixed with 5b71b91.
